include/mega/attrmap.h

```diff
@@ -27,7 +27,29 @@
 namespace mega {
 
 // maps attribute names to attribute values
-typedef map<nameid, string> attr_map;
+struct attr_map : map<nameid, string> 
+{
+    attr_map() {}
+    
+    attr_map(nameid key, string value)
+    {
+        (*this)[key] = value;
+    }
+
+    attr_map(map<nameid, string>&& m)
+    {
+        m.swap(*this);
+    }
+
+    void applyUpdates(const attr_map& updates)
```

Comment:
Yes, I was thinking it might be useful to be able to use that outside AttrMap, but I haven't used it that way, and it's a bit cleaner as an AttrMap member.  I've moved it.